eventmetrics/parsers/provide.go

```diff
@@ -256,46 +336,104 @@ func createBootDurationCallback(m Measures) (func(interpreter.Event, float64), e
 	}, nil
 }
 
-func createRebootToManageableCallback(m Measures) (func(interpreter.Event, interpreter.Event, float64), error) {
-	if m.RebootToManageableHistogram == nil {
-		return nil, errNilRebootHistogram
+func createTimeElapsedCallback(m Measures, name string) (func(interpreter.Event, interpreter.Event, float64), error) {
+	if m.TimeElapsedHistograms == nil {
+		return nil, errNilHistogram
+	}
+
+	if _, found := m.TimeElapsedHistograms[name]; !found {
+		return nil, errNilHistogram
 	}
 
 	return func(currentEvent interpreter.Event, startingEvent interpreter.Event, duration float64) {
 		labels := getTimeElapsedHistogramLabels(currentEvent)
-		m.RebootToManageableHistogram.With(labels).Observe(duration)
+		histogram := m.TimeElapsedHistograms[name]
+		histogram.With(labels).Observe(duration)
 	}, nil
 }
 
-func createEventValidator(config RebootParserConfig) validation.Validator {
-	config = checkTimeValidations(config)
+func createEventValidator(config EventValidationConfig) (validation.Validator, error) {
+	validationType := enums.ParseValidationType(config.Key)
```

Comment:
Instead of parsing here, you can have an enum as a part of the config struct directly...I think all you need is an UnmarshalText() function like this example:
https://www.geeksforgeeks.org/time-time-unmarshaltext-function-in-golang-with-examples/#:~:text=The%20UnmarshalText()%20function%20in,defined%20under%20the%20time%20package.

Reply:
super cool!